pkg/agent/svid/rotator_test.go

```diff
@@ -96,24 +156,20 @@ func TestRotator(t *testing.T) {
 				errCh <- rotator.Run(ctx)
 			}()
 
-			// Wait for the initial rotation check to finish
-			select {
-			case <-time.After(time.Minute):
-				t.Fatal("timed out waiting for rotation check to finish")
-			case <-rotationDone:
-				clk.WaitForAfter(time.Minute, "timed out waiting for rotation loop to start waiting")
-			}
-
 			// Optionally advance the clock by the specified amount
-			// and wait for another rotation check to finish.
+			// before waiting for the rotation check to finish.
 			if tt.checkAfter != 0 {
 				require.Greater(t, tt.checkAfter, DefaultRotatorInterval, "checkAfter should be larger than the default rotator interval")
 				clk.Add(tt.checkAfter)
+			}
+
+			// Wait for the rotation check to finish
+			if tt.shouldRotate || tt.shouldReattest {
 				select {
 				case <-time.After(time.Minute):
-					t.Fatal("timed out waiting for rotation check to finish after clock adjustment")
+					t.Fatal("timed out waiting for rotation check to finish")
 				case <-rotationDone:
```

Comment:
This `rotationDone` previously was always set because finished hook was called multiple times before any rotation was actually completed.